src/main/java/org/broadinstitute/hellbender/tools/copynumber/PostprocessGermlineCNVCalls.java

```diff
@@ -1,221 +1,605 @@
 
-    private static final String COMMENT_PREFIX = "@";
+    public static final String CALLS_SHARD_PATH_LONG_NAME = "calls-shard-path";
+    public static final String MODEL_SHARD_PATH_LONG_NAME = "model-shard-path";
+    public static final String CONTIG_PLOIDY_CALLS_LONG_NAME = "contig-ploidy-calls";
+    public static final String SAMPLE_INDEX_LONG_NAME = "sample-index";
+    public static final String OUTPUT_INTERVALS_VCF_LONG_NAME = "output-genotyped-intervals";
+    public static final String OUTPUT_SEGMENTS_VCF_LONG_NAME = "output-genotyped-segments";
+    public static final String DRY_RUN_LONG_NAME = "dry-run";
+    public static final String AUTOSOMAL_REF_COPY_NUMBER_LONG_NAME = "autosomal-ref-copy-number";
+    public static final String ALLOSOMAL_CONTIG_LONG_NAME = "allosomal-contig";
 
     @Argument(
-            doc = "List of paths to GermlineCNVCaller call directories, sorted in order of the intervals contained.",
-            fullName = CHUNK_PATH_LONG_NAME,
+            doc = "List of paths to GermlineCNVCaller call directories.",
+            fullName = CALLS_SHARD_PATH_LONG_NAME,
             minElements = 1
     )
-    private List<File> orderedChunkDirectoryList = null;
+    private List<File> unsortedCallsShardPaths;
+
+    @Argument(
+            doc = "List of paths to GermlineCNVCaller model directories.",
+            fullName = MODEL_SHARD_PATH_LONG_NAME,
+            minElements = 1,
+            optional = true
+    )
+    private List<File> unsortedModelShardPaths;
+
+    @Argument(
+            doc = "Path to contig-ploidy calls directory (output of DetermineGermlineContigPloidy).",
+            fullName = CONTIG_PLOIDY_CALLS_LONG_NAME,
+            optional = true
+    )
+    private File contigPloidyCallsPath = null;
 
     @Argument(
-            doc = "Name of the sample directory (must be contained in all call directories).",
-            fullName = SAMPLE_DIRECTORY_LONG_NAME
+            doc = "Sample index in the call-set (must be contained in all shards).",
+            fullName = SAMPLE_INDEX_LONG_NAME,
+            minValue = 0
     )
-    private String sampleDirectoryName = null;
+    private int sampleIndex;
 
     @Argument(
-            doc = "Output VCF file.",
-            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
-            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+            doc = "Reference copy-number on autosomal intervals.",
+            fullName = AUTOSOMAL_REF_COPY_NUMBER_LONG_NAME,
+            minValue = 0
     )
-    private File outputFile = null;
+    private int refAutosomalCopyNumber = 2;
+
+    @Argument(
+            doc = "Contigs to treat as allosomal (hence, set their reference copy-number allele according to " +
+                    "their neutral baseline ploidy state).",
+            fullName = ALLOSOMAL_CONTIG_LONG_NAME,
+            optional = true
```

Comment:
Would the following work if `allosomal_contigs` if an optional `Array[String]`?
```
        gatk --java-options "-Xmx${command_mem_mb}m" PostprocessGermlineCNVCalls \
            $calls_args \
            $model_args \
            ${"--allosomal-contig " + ${sep=" --allosomal-contig " allosomal_contigs}} \
            --autosomal-ref-copy-number ${ref_copy_number_autosomal_contigs} \
            --contig-ploidy-calls extracted-contig-ploidy-calls \
            --sample-index ${sample_index} \
            --output-genotyped-intervals ${genotyped_intervals_vcf_filename} \
            --output-genotyped-segments ${genotyped_segments_vcf_filename}
```

I'm not quite familiar with Cromwell's handling of optional types, but I suppose if `allosomal_contigs` is undefined, then `${sep=" --allosomal-contig " allosomal_contigs}}` is undefined, and therefore the whole thing `${"--allosomal-contig " + ${sep=" --allosomal-contig " allosomal_contigs}}`, so that we get no extra command line argument (as expected)?

Reply:
Not sure if that would work, maybe you can try?  You could always use `is_defined(allosomal_contigs)` to determine whether or not to print the initial `--allosomal_contigs`.  Perhaps also try searching the WDL forums to see if there is a standard way to do this.